ocrd_eval.md

```diff
@@ -0,0 +1,537 @@
+Letter Accuracy can be calculated as follows:
+
+Let $|L_{GT}|$ be the number of relevant letters in the ground truth, $|L_{r}|$ the number of recognized letters, then
+
+$LA = 1 - \frac{|L_{GT}| - |L_{r}|}{|L_{GT}|}$
+
+##### Flexible Character Accuracy Measure
+
+The Flexible Character Accuracy (FCA) measure has been introduced to mitigate a major drawback of CER:
+CER (if applied naively by comparing concatenated page-level texts) is heavily dependent on the reading order an OCR engine detects.
+Thus, where text blocks are rearranged or merged, no suitable text alignment can be made, so CER is very low,
+even if single characters, words and even lines have been perfectly recognized.
+
+FCA avoids this by splitting the recognized text and GT into lines and, if necessary, sub-line chunks,
+finding pairs that align maximally until only unmatched lines remain (which must be treated as errors),
+and measuring average CER of all pairs.
+
+The algorithm can be summarized as follows:
+
+> 1. Split both input texts into text lines
+> 2. Sort the GT lines by length  
+>      (in descending order)
+> 3. For the top GT line, find the best fully or partially matching OCR line  
+>      (by lowest edit distance and highest coverage)
+> 4. If full match (i.e. full length of line)
+>     a. Mark as done and remove line from both lists
+>     b. Else mark matching part as done,  
+>         then cut off unmatched part and add to respective list of text lines; resort
+> 5. If any more lines available repeat step 3
+> 6. Count remaining unmatched lines as insertions or deletions (depending on origin – GT or OCR)
+> 7. Calculate the (micro-)average CER of all marked pairs and return as overall FCER
+
+(paraphrase of C. Clausner, S. Pletschacher and A. Antonacopoulos / Pattern Recognition Letters 131 (2020) 390–397, p. 392)
+
+#### Layout Evalutation
+
+##### Reading Order Evaluation
+
+[Clausner, Pletschacher and Antonacopoulos 2013](https://www.primaresearch.org/www/assets/papers/ICDAR2013_Clausner_ReadingOrder.pdf) propose a way of evaluating reading order by classifying relations between two regions.
+For both the ground truth and the detected reading order the type of relation between two regions are calculated and then compared.
+The authors introduce a penalty matrix in which a penality (as integer) for a misclassified relation is given.
+The more the detected relation differs from the relation in the ground truth, the higher the penalty.
+
+If for example the relation given in the ground truth is "Somewhere after (but unordered group involved)" and the detected relation is "directly before" the penalty is lower (`10`) than as if the ground truth relation was "directly after" (`40`) because the latter is more specific than the former.
+
+To calculate the success measure $s$ of the detected reading order, first all penalties obtained from comparing all GT to detected relations are summed up ($e$) and the error value at a success rate of 50% is determined by
```

Comment:
note: we are not talking about penalties here, but costs (already including weights)

also, $e_50$ is itself not a success score, but a (hypothetical) cost (or error rate)

```suggestion
To calculate the success measure $s$ of the detected reading order, first the costs obtained from comparing all GT to all detected relations are summed up ($e$).
Then this error value is normalised by the hypothetical error value at 50% agreement ($e_{50}$):
```